### DISCUSSION TITLE
How to compile?

### DISCUSSION THREAD

**Discussion by jardicc** [uNetworking/uWebSockets.js#211] [Q&A] [answered]
I want to make some changes in C++ source code for my own needs only. How can I compile a solution? I tried to use node-gyp add bindings and configuration but it says "can't include file App.h, no such a file" Or should I even use node-gyp? Or do I need to add files somewhere/somehow?

**Comment by dioptre** [comment, marked as answer]
just run 'make'

**Comment by jardicc** [comment]
Where? How? In what tool? Just "make" or make with some arguments? I am on Windows, do I need any prerequisites?

**Comment by jardicc** [comment]
I used chocolatey to install "make" and I see this error:
```
PS A:\BR\experiments\uws2\uWebSockets.js> make -f build.c
build.c:6: *** missing separator.  Stop.
```

and also:
```
PS A:\BR\experiments\uws2\uWebSockets.js> make
cc build.c
process_begin: CreateProcess(NULL, cc build.c, ...) failed.
make (e=2): SystÚm nem¨×e nalÚzt uvedenř soubor. (system can't find specified file)
make: *** [Makefile:2: Windows] Error 2
```

**Comment by jardicc** [comment]
It is not. But can I have a question somewhere?

**Comment by jardicc** [comment]
Is somewhere a proper place where could I get an answer? Because I couldn't find anything here although this project is very well commented and has good documentation.

**Comment by Awk34** [comment]
StackOverflow.com ?
